addons/website_slides/static/src/xml/slide_quiz.xml

```diff
@@ -96,26 +96,32 @@
-                                <h1 class="mt-3 display-4 font-weight-bold">Amazing!</h1>
+                                <h1 class="o_wslides_quiz_modal_title mt-3 display-4 font-weight-bold">Amazing!</h1>
                                 <div class="pb-3">
-                                    <h4 class="pb-2 d-flex">You gained <span class="badge badge-pill badge-success text-white font-weight-bold ml-2 mr-1"><t t-esc="widget.quiz.quizKarmaWon"/> XP</span> !</h4>
-                                    <div class="mt-3 mb-4">
-                                        <div class="progress" style="height: 8px">
-                                            <div class="progress-bar no-transition" role="progressbar" t-att-aria-valuenow="widget.quiz.rankProgress.oldProgress" aria-valuemin="0" aria-valuemax="100"
-                                                t-attf-style="width: #{widget.quiz.rankProgress.oldProgress}%"/>
+                                    <h4 class="o_wslides_xp_gained pb-2 d-flex" style="opacity: 0;">You gained <span class="badge badge-pill badge-success text-white font-weight-bold ml-2 mr-1"><t t-esc="widget.quiz.quizKarmaWon"/> XP</span> !</h4>
+                                    <div class="mt-5 mb-4">
+                                        <div class="progress">
+                                            <div class="progress-bar no-transition" role="progressbar" t-att-aria-valuenow="widget.quiz.rankProgress.previous_rank.progress" aria-valuemin="0" aria-valuemax="100"
+                                                t-attf-style="width: #{widget.quiz.rankProgress.previous_rank.progress}%"/>
+                                            <div class="progress-bar-tooltip" data-toggle="tooltip" data-placement="top" t-att-title="widget.quiz.rankProgress.new_rank.karma" />
                                         </div>
-                                        <small t-if="widget.quiz.rankProgress.oldLowerBound" class="float-left text-primary font-weight-bold o_wslides_rank_lower_bound"><t t-esc="widget.quiz.rankProgress.oldLowerBound"/></small>
-                                        <small t-if="widget.quiz.rankProgress.oldUpperBound" class="float-right font-weight-bold o_wslides_rank_upper_bound"><t t-esc="widget.quiz.rankProgress.oldUpperBound"/></small>
+                                        <small t-if="widget.quiz.rankProgress.previous_rank.lower_bound" class="float-left text-primary font-weight-bold o_wslides_rank_lower_bound"><t t-esc="widget.quiz.rankProgress.previous_rank.lower_bound"/></small>
+                                        <small t-if="widget.quiz.rankProgress.previous_rank.upper_bound || widget.quiz.rankProgress.last_rank" class="float-right font-weight-bold o_wslides_rank_upper_bound"><t t-esc="(!widget.quiz.rankProgress.level_up and widget.quiz.rankProgress.last_rank) ? widget.quiz.rankProgress.new_rank.karma : widget.quiz.rankProgress.previous_rank.upper_bound"/></small>
```

Comment:
This is going a lot to the right.
Maybe indent to lower scrolling?

Also this:
`(!widget.quiz.rankProgress.level_up and widget.quiz.rankProgress.last_rank) ? widget.quiz.rankProgress.new_rank.karma : widget.quiz.rankProgress.previous_rank.upper_bound`

Could be computed in the python if you can find a nice name for it.
Like "upper_bound_message" or something ?

Reply:
As discussed I removed the display of the upper_bound when the last rank is reached since the karma is already displayed in the tooltip.